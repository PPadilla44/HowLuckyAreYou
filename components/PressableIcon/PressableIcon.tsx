import { Pressable } from 'react-native'
import React, { FC } from 'react'
import { Icon } from '../Themed'

interface Props {
    callBack: () => any,
    name: string,
    size: number,
    type: string
}

const PressableIcon: FC<Props> = ({ callBack, name, size, type }) => {
    return (
        <Pressable
            testID='pressableIconBtn'
            onPress={callBack}
            style={({ pressed }) => ({
                opacity: pressed ? 0.3 : 1
            })}>
            <Icon
                testID='pressableIcon'
                name={name}
                type={type}
                size={size}
            />
        </Pressable>
    )
}

export default PressableIcon