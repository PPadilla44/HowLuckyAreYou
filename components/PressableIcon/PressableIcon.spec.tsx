import { cleanup, fireEvent, render, RenderAPI } from "@testing-library/react-native";
import PressableIcon from "./PressableIcon";

describe('<PressableIcon', () => {

    afterEach(cleanup);

    let wrapper: RenderAPI;
    let mockFn = jest.fn()
    beforeEach(() => {
        wrapper = render(<PressableIcon name="gear" size={40} type="evilicon" callBack={mockFn} />)
    })

    it("renders correctly", () => {
        wrapper.getByTestId("pressableIconBtn");
        wrapper.getByTestId("pressableIcon");
        wrapper.container.findByProps({name: "gear"});
        wrapper.container.findByProps({ type: "evilicon" })
        wrapper.container.findByProps({ size: 40 })
    });
    
    it("runs call back function on button press", () => {
        const btn = wrapper.getByTestId("pressableIconBtn");
        fireEvent.press(btn);
        wrapper.debug()
        expect(mockFn).toBeCalled();
    });


});