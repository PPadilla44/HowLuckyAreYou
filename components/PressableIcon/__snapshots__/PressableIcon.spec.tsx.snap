// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PressableIcon Matches Snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="pressableIconBtn"
>
  <View
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <View
      testID="pressableIcon"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 40,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "fontFamily": "evilicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          ï„˜
        </Text>
      </View>
    </View>
  </View>
</View>
`;
